### ISSUE TITLE
Many tests should be renamed

### ISSUE THREAD

**Issue by Micket** [oofem/oofem#2]
Many tests have names such as "tmpatchXX", which includes no additional information, as it's already known to be part of the tm directory, and almost everything is named "patch".

I think any test intended to test a material model should be named consistently by the material model it intends to test, simply:
mat_{material_name}.in
{element_name}.in
If multiple test are required, to fully test a component, numbering can be used.

Many tests already contain a good name, e.g. spring, beam, and some lack a bit of information (axisymm01? there are many axisymmetric elements).
In some cases, I think some tests may be quite redundant; 6 different cases for axisymm for example.
Perhaps some unit-testing could take the place of such in-depth testing.

If we find that tests that doesn't really test any new component, they should be reworked, split, or even removed as to not clutter up.

If needed, we can add the folder name as a prefix for when generating the test names in CMake, such that `sm/foo.in` becomes `sm_foo`

**Comment by bpatzak** [comment]
I agree, just want to point out, that in many cases, it is difficult to figure out, what is actually tested, as it is always a combination of node+element+material+BCs+solver functionality, we are not yet able to test separately material models, elements, etc. 
These separate unit tests would be very beneficial, but we need to develop templates for testing individual materials, elements, etc. Such a template can, for example, accept strain history, call material model and compare results with expected ones (stresses, internal variables,...)

**Comment by bpatzak** [comment]
Does anyone has a recommendation for suitable c++ unit test framework?

**Comment by Micket** [comment]
I don't really have any experience with testing, but I think either of Catch, Boost Test,  or Google Test would be a good fit. Catch2 has the advantage of (single) header only.
